app/uk/gov/hmrc/teamsandrepositories/models/BuildJob.scala

```diff
@@ -22,54 +22,96 @@ import uk.gov.hmrc.mongo.play.json.formats.MongoJavatimeFormats
 
 object BuildJob {
 
   val mongoReadFormat: Reads[BuildJob] =
     ((__ \ "service").read[String]
       ~ (__ \ "jenkinsURL").read[String]
-      ~ (__ \ "latestBuild").readNullable(BuildJobBuildData.mongoFormat.reads)
+      ~ (__ \ "latestBuild").readNullable(BuildData.mongoFormat.reads)
       ) (apply _)
 
   val mongoWriteFormat: Writes[BuildJob] =
     ((__ \ "service").write[String]
       ~ (__ \ "jenkinsURL").write[String]
-      ~ (__ \ "latestBuild").writeNullable(BuildJobBuildData.mongoFormat.writes)
+      ~ (__ \ "latestBuild").writeNullable(BuildData.mongoFormat.writes)
       ) (unlift(unapply))
 
   val mongoFormat: Format[BuildJob] = Format(mongoReadFormat, mongoWriteFormat)
 
   val apiWrites: Writes[BuildJob] =
-    ( (__ \ "service"   ).write[String]
-    ~ (__ \ "jenkinsURL").write[String]
-    ~ (__ \ "latestBuild").writeNullable(BuildJobBuildData.apiWrites)
-    )(unlift(unapply))
+    ((__ \ "service").write[String]
+      ~ (__ \ "jenkinsURL").write[String]
+      ~ (__ \ "latestBuild").writeNullable(BuildData.apiWrites)
+      ) (unlift(unapply))
+
+  val jenkinsBuildJobReads: Reads[BuildJob] = (
+    (__ \ "name").read[String]
+      ~ (__ \ "url").read[String]
+      ~ (__ \ "lastBuild").readNullable[BuildData](BuildData.jenkinsReads)
+    ) (apply _)
 }
 
-case class BuildJobBuildData(
-                      number: Int,
-                      url: String,
-                      timestamp: Instant,
-                      result: Option[BuildResult]
-                    )
-object BuildJobBuildData {
+object JenkinsObject {
 
-  val mongoFormat: OFormat[BuildJobBuildData] =
-    ((__ \ "number").format[Int]
-      ~ (__ \ "url").format[String]
-      ~ (__ \ "timestamp").format(MongoJavatimeFormats.instantFormat)
-      ~ (__ \ "result").formatNullable[BuildResult]
-      ) (apply, unlift(unapply))
+  private lazy val folderReads: Reads[JenkinsFolder] = (
+    (__ \ "name").read[String]
+      ~ (__ \ "url").read[String]
+      ~ (__ \ "jobs").lazyRead(Reads.seq[JenkinsObject](jenkinsObjectReads))
+    ) (JenkinsFolder)
+  private val pipelineReads: Reads[PipelineJob] = (
+    (__ \ "name").read[String]
+      ~ (__ \ "url").read[String]
+    ) (PipelineJob)
 
-  val apiWrites: Writes[BuildJobBuildData] =
-    ((__ \ "number").write[Int]
-      ~ (__ \ "url").write[String]
-      ~ (__ \ "timestamp").write[Instant]
-      ~ (__ \ "result").writeNullable[BuildResult]
-      ) (unlift(unapply))
+  implicit val jenkinsObjectReads: Reads[JenkinsObject] = json =>
+    json \ "_class" match {
+      case JsDefined(JsString("com.cloudbees.hudson.plugins.folder.Folder")) => folderReads.reads(json)
+      case JsDefined(JsString("hudson.model.FreeStyleProject")) => BuildJob.jenkinsBuildJobReads.reads(json)
+      case JsDefined(JsString("org.jenkinsci.plugins.workflow.job.WorkflowJob")) => pipelineReads.reads(json)
+    }
 }
+
+case class JenkinsBuildJobsWrapper(jobs: Seq[JenkinsObject])
```

Comment:
The `Wrapper` suffix seems superfluous. To match the contained data, maybe `JenkinsObjects` would be better?